dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,228 @@
+
+This creates a chain that is static frame -> dynamic frame -> static frame.
+
+In this chain, only the dynamic frame will be updated but that update would cascade to any data that
+references the final static frame.
+
+In XVIZ a static entry would be sent in a _PERSISTENT_ state_update message. Dyanmic links and poses
+would come from a _COMPLETE_ or _INCREMENTAL_ message.
+
+5. Missing or invalid transform chains
+
+If there are holes in the transform chain then the stream should be marked with an error such that
+the viewing client can communicate the nature of the error to the users.
+
+## Message combinations
+
+State update messages have COMPLETE, INCREMENTAL and PERSISTENT types.
+
+### COMPLETE and INCREMENTAL
+
+These modes fundamentally deal with creating and adding data that is subject to the TIME_WINDOW
+configuration. The intention is that dynamic poses and links that are updated frequently, defined as
+being within the TIME_WINOW, would be sent in either of these messages.
+
+The semantics of _INCREMENTAL_ are at the stream set level, so it is worth noting that transforms
+are not combined but an _INCREMENTAL_ message would simply replace the active pose at that
+timestamp.
+
+Links should be avoided in these messages as a situation where the link would be only shortly lived
+is unclear in the current use-cases. They can be defined but will only be present for the duration
+of TIME_WINDOW.
+
+Keep in mind that these messages may also be purged as the assumption in an XVIZ client is that a
+minimal amount of state is necessary.
+
+### PERSISTENT
+
+PERSISTENT messages is where most links should be defined as they setup a structure that changes
+very little if at all. Static poses for a link, like those from a platform to a sensor, should also
+be defined in this message type.
+
+PERSISTENT messages define state that is not subject to the TIME_WINDOW and also less likely to be
+purged. The messages are still subject to a timestamp and therefore care must be taken to ensure
+this data does not grow unbounded in the application state.
+
+## Streetscape.gl implementation details
+
+The streetscape.gl implementation currently manages the transform resolution through a single
+function, resolveCoordinateTransform(), which is given access to the the internal data frame and the
+the stream metadata. With that information it can process the _links_ transform chain to compute the
+affective matrix for rendering.
+
+We may want to build a transform cache if necessary, but that is a performance implmentation detail.
+
+# Follow-up proposals
+
+The following are proposals that are related to this but are not required for this proposal to be
+accepted. They are mentioned only to fully understand the scope, influence, and considerations that
+have helped contextualize and shape this proposal.
+
+- Making timestamps optional on the streamset
+- Support for linear interpolation on poses
```

Comment:
If we are not going to say how interpolation works now, I would be more explicit about how this proposal handles transforms that change over time.   It's seems like there are two cases really:

 - "persistent" transforms - for these you use the most recent transform relative to the timestamp. Presumably these would be sent in a `PERSISTENT` (which is how you know to use this policy)
 - "dynamic" transforms - these would come in a `COMPLETE` or `INCREMENTAL` message.  Most users would limited to know extrapolation between transform updates.

Reply:
Ok.

`extrapolation` really?  Is that just a linear function from the last 2 samples or 'n' samples?

Fundamentally, for the XVIZ case, this seems related to the XVIZStreamBuffer (a new name coming for that) maybe something that sits on top of the XVIZStreamBuffer and allows customization for how this is calculated.    Another OSS partner mentioned this as a P0 for them.

Reply:
I don't think I captured your bullets entirely. Lets sync on this